### PR TITLE
Add processor pools as resource pools to inventory collection

### PR DESCRIPTION
**Pull request by alizapeikes**
Depends on:
- https://github.com/IBM-Cloud/ibm-cloud-sdk-ruby/pull/70
- https://github.com/ManageIQ/manageiq/pull/22228

### CODE REVIEW

**Comment by agrare** [comment]
@miq-bot cross-repo-tests https://github.com/ManageIQ/manageiq/pull/22228

**Comment by agrare** [comment]
@alizapeikes seems like this is primarily held up by https://github.com/IBM-Cloud/ibm-cloud-sdk-ruby/pull/70 is that right?

**Comment by alizapeikes** [comment]
> @alizapeikes seems like this is primarily held up by [IBM-Cloud/ibm-cloud-sdk-ruby#70](https://github.com/IBM-Cloud/ibm-cloud-sdk-ruby/pull/70) is that right?

Yes. @jaywcarman is also working on fixing the powerVS refresher spec VCR so we can include resource_pools in the testing.

**Comment by miyamotoh** [comment]
With this PR, I can get my test pool from a PowerVS instance, and get it show up properly in a Resource Pool Summary report.
image
@agrare what to do to retry CI checks?

File: spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/refresher_spec.rb [hunk 1]
```diff
@@ -112,14 +112,13 @@ def assert_specific_flavor
...
       vm = ems.vms.find_by(:name => instance_name)
-      placement_group = ems.placement_groups.find_by(:name => "test-placement-group-affinity")
       expect(vm).to have_attributes(
         :location           => "unknown",
         :name               => instance_name,
         :description        => "PVM Instance",
         :vendor             => "ibm_power_vs",
         :power_state        => "on",
-        :placement_group_id => placement_group.id,
+        :placement_group_id => nil,
```
Comment (agrare) [inline spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/refresher_spec.rb:121]: Hey so my only concern here is that we're no longer testing the placement_group lazy find here https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/431/files#diff-f985d1bd514a044797c4062d1d8d94ff3da33e97f6bfa8d182955661184e7d45R64
We should also be testing the resource group lazy find https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/431/files#diff-f985d1bd514a044797c4062d1d8d94ff3da33e97f6bfa8d182955661184e7d45R65

Comment (jaywcarman) [reply spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/refresher_spec.rb]: Good catch. I switched `assert_specific_vm` to use a VM that is still a member of a placement group so we can keep the lazy find coverage.

For the resource group lazy find we'll need to record another VCR cassette with a shared processor pool (the current cassette doesn't have any). fyi @miyamotoh

**Comment by agrare** [comment]
Closing in favor of https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/443

**Pull request closed without merging**